### PR TITLE
Prepare release for Chocolatey

### PR DESCRIPTION
**Pull request by skylarnam**
- Added a choco pack script
- Added nuspec, verification, and props
- side: Added a microbuild cleanup task

### CODE REVIEW

File: pkg/VSConfigFinder/build/VSConfigFinder.props [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <VSWhereDir>$(MSBuildThisFileDirectory)..\tools\</VSWhereDir>
```
Comment (tydunkel) [inline pkg/VSConfigFinder/build/VSConfigFinder.props]: VSWhere?

File: pkg/VSConfigFinder/VSConfigFinder.nuspec [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    <tags>vs vs2022 visualstudio</tags>
+    <developmentDependency>true</developmentDependency>
+    <projectSourceUrl>https://github.com/Microsoft/VSConfigFinder/tree/$CommitId$</projectSourceUrl>
+    <packageSourceUrl>https://github.com/Microsoft/VSConfigFinder/tree/$CommitId$/pkg/VSConfigFinder</packageSourceUrl>
+    <docsUrl>https://github.com/microsoft/VSConfigFinder/blob/main/README.md</docsUrl>
+    <bugTrackerUrl>https://github.com/Microsoft/VSConfigFinder/issues</bugTrackerUrl>
+    <releaseNotes>https://github.com/Microsoft/VSConfigFinder/releases/tag/$Tag$</releaseNotes>
+  </metadata>
+  <files>
+    <file src="build\VSConfigFinder.props" target="build\"/>
```
Comment (tydunkel) [inline pkg/VSConfigFinder/VSConfigFinder.nuspec]: Why are we including the props file?

File: pkg/VSConfigFinder/VSConfigFinder.nuspec [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (tydunkel) [inline pkg/VSConfigFinder/VSConfigFinder.nuspec:1]: The more modern approach is to set this up in the csproj

File: build.yml [hunk 4]
```diff
@@ -32,6 +32,11 @@ steps:
     publishWebProjects: false
     zipAfterPublish: false
 
+- script: |
+    choco pack pkg\vswhere\VSConfigFinder.nuspec --out "bin\${{ parameters.BuildConfiguration }}" --version "%NBGV_NuGetPackageVersion%" "Configuration=${{ parameters.BuildConfiguration }}" "CommitId=$(Build.SourceVersion)" "Tag=$(Build.BuildNumber)"
```
Comment (tydunkel) [inline build.yml]: vswhere?

File: pkg/VSConfigFinder/VSConfigFinder.nuspec [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
```
Comment (tydunkel) [inline pkg/VSConfigFinder/VSConfigFinder.nuspec:2]: Do we have to create a nupkg to create a choco package? What do we need to produce to publish to WinGet?

**Review by tydunkel — CHANGES_REQUESTED** [review]

File: pkg/VSConfigFinder/VSConfigFinder.nuspec (hunk 3 above)
Comment (heaths) [reply pkg/VSConfigFinder/VSConfigFinder.nuspec:1]: Yes, but having gone down this route, you have to fully commit to the new SDK-style project. Is that the case with the .csproj? If you still use the old-style .NET Framework-style projects, it won't work. By the time you pull in all the right props/targets yourself to try to make it work, there are so many conflicts and misaligned targets it's a mess to untangle.

File: pkg/VSConfigFinder/VSConfigFinder.nuspec (hunk 5 above)
Comment (skylarnam) [reply pkg/VSConfigFinder/VSConfigFinder.nuspec:2]: For WinGet, we use what's published on GitHub. Nuspec is required to create a choco package (https://docs.chocolatey.org/en-us/create/create-packages#okay-how-do-i-create-packages).

Comment (tydunkel) [reply pkg/VSConfigFinder/VSConfigFinder.nuspec:2]: Taking a look, choco expects a nupkg so this should be good.

File: pkg/VSConfigFinder/VSConfigFinder.nuspec (hunk 2 above)
Comment (skylarnam) [reply pkg/VSConfigFinder/VSConfigFinder.nuspec]: Hm. I thought vswhere defined vswheredir in order to use it in the release, but I don't think it does. I think I can remove it?

Comment (skylarnam) [reply pkg/VSConfigFinder/VSConfigFinder.nuspec]: Found it on vswhere's wiki:

> For ease of use, installing the package into one or more projects in a solution will automatically include vswhere.props which defines a VSWhereDir MSBuild property you can use in your project however you like.

Comment (skylarnam) [reply pkg/VSConfigFinder/VSConfigFinder.nuspec]: I don't think it really applies to our use case, so I'll still remove it.

File: pkg/VSConfigFinder/VSConfigFinder.nuspec [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    <version>$Version$</version>
+    <title>Visual Studio Configuration Finder</title>
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <iconUrl>https://go.microsoft.com/fwlink/?linkid=839265</iconUrl>
+    <licenseUrl>https://github.com/Microsoft/VSConfigFinder/tree/$CommitId$/LICENSE.txt</licenseUrl>
+    <projectUrl>https://github.com/Microsoft/VSConfigFinder</projectUrl>
+    <description>Locate Visual Studio installation configuration files (.vsconfigs) downstream recursively and merge them altogether</description>
+    <summary>Locate Visual Studio installation configuration files (.vsconfigs) downstream recursively and merge them altogether</summary>
```
Comment (tydunkel) [inline pkg/VSConfigFinder/VSConfigFinder.nuspec]: nit: I think this should say `all together`

```suggestion
    <summary>Locate Visual Studio installation configuration files (.vsconfigs) downstream recursively and merge them all together</summary>
```

File: pkg/VSConfigFinder/VSConfigFinder.nuspec [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    <tags>vs vs2022 visualstudio</tags>
+    <developmentDependency>true</developmentDependency>
+    <projectSourceUrl>https://github.com/Microsoft/VSConfigFinder/tree/$CommitId$</projectSourceUrl>
+    <packageSourceUrl>https://github.com/Microsoft/VSConfigFinder/tree/$CommitId$/pkg/VSConfigFinder</packageSourceUrl>
+    <docsUrl>https://github.com/microsoft/VSConfigFinder/blob/main/README.md</docsUrl>
+    <bugTrackerUrl>https://github.com/Microsoft/VSConfigFinder/issues</bugTrackerUrl>
+    <releaseNotes>https://github.com/Microsoft/VSConfigFinder/releases/tag/$Tag$</releaseNotes>
+  </metadata>
+  <files>
+    <file src="..\..\bin\$Configuration$\VSConfigFinder.exe" target="tools\"/>
```
Comment (tydunkel) [inline pkg/VSConfigFinder/VSConfigFinder.nuspec]: I don't think this path is correct. Don't we need to pull from the `publish` dir?

**Review by tydunkel — CHANGES_REQUESTED** [review]

**Comment by skylarnam** [comment]
image

**Review by tydunkel — APPROVED** [review]

**Pull request merged**